celery/utils/timer2.py

```diff
@@ -93,8 +93,9 @@ def run(self):
     def stop(self):
         self._is_shutdown.set()
         if self.running:
-            self._is_stopped.wait()
-            self.join(THREAD_TIMEOUT_MAX)
+            if not self._is_stopped:
```

Comment:
Why is this a boolean on PyPy?

Reply:
@mjhoffman65 Ping?

Reply:
I'm suspicious about this change. If this is falsey, we're going to try an `.wait()` on it which seems odd since that probably implies that it's some garbage object. I don't think `Event`s should be treated like this, and if it is a `boolean` in the `pypy3` case, then we're going to do `bool.wait()` which seems wrong.